nova/virt/vmwareapi/vm_util.py

```diff
@@ -1731,6 +1784,7 @@ def create_vm(session, instance, vm_folder, config_spec, res_pool_ref):
     return task_info.result
 
 
+@vm_ref_cache_heal_from_instance
 def destroy_vm(session, instance, vm_ref=None):
```

Comment:
If I see this correctly, the decorator will not work on this function as it handles `Exception` and then just logs and doesn't reraise. Same for `mark_vm_as_template()`.

Reply:
Ah, I missed that, I'll fix `destroy_vm`, and it doesn't make applying it to `mark_vm_as_template`, as the template won't have the uuid of the instance.